hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/MariaDBLegacyDialect.java

```diff
@@ -0,0 +1,183 @@
+import org.hibernate.engine.jdbc.dialect.spi.DialectResolutionInfo;
+import org.hibernate.engine.jdbc.env.spi.IdentifierCaseStrategy;
+import org.hibernate.engine.jdbc.env.spi.IdentifierHelper;
+import org.hibernate.engine.jdbc.env.spi.IdentifierHelperBuilder;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.query.spi.QueryEngine;
+import org.hibernate.sql.ast.SqlAstTranslator;
+import org.hibernate.sql.ast.SqlAstTranslatorFactory;
+import org.hibernate.sql.ast.spi.StandardSqlAstTranslatorFactory;
+import org.hibernate.sql.ast.tree.Statement;
+import org.hibernate.sql.exec.spi.JdbcOperation;
+import org.hibernate.tool.schema.extract.internal.SequenceInformationExtractorMariaDBDatabaseImpl;
+import org.hibernate.tool.schema.extract.spi.SequenceInformationExtractor;
+import org.hibernate.type.StandardBasicTypes;
+
+/**
+ * A {@linkplain Dialect SQL dialect} for MariaDB
+ *
+ * @author Vlad Mihalcea
+ * @author Gavin King
+ */
+public class MariaDBLegacyDialect extends MySQLLegacyDialect {
+	private static final DatabaseVersion VERSION5 = DatabaseVersion.make( 5 );
+	private static final DatabaseVersion VERSION57 = DatabaseVersion.make( 5, 7 );
+
+	public MariaDBLegacyDialect() {
+		this( DatabaseVersion.make( 5 ) );
+	}
+
+	public MariaDBLegacyDialect(DatabaseVersion version) {
+		super(version);
+	}
+
+	public MariaDBLegacyDialect(DialectResolutionInfo info) {
+		super(info);
+	}
+
+	@Override
+	public DatabaseVersion getMySQLVersion() {
+		return getVersion().isBefore( 5, 3 )
+				? VERSION5
+				: VERSION57;
+	}
+
+	@Override
+	public NationalizationSupport getNationalizationSupport() {
+		return NationalizationSupport.IMPLICIT;
+	}
+
+	@Override
+	public void initializeFunctionRegistry(QueryEngine queryEngine) {
+		super.initializeFunctionRegistry(queryEngine);
+
+		if ( getVersion().isSameOrAfter( 10, 2 ) ) {
+			CommonFunctionFactory commonFunctionFactory = new CommonFunctionFactory( queryEngine );
+			commonFunctionFactory.windowFunctions();
+			commonFunctionFactory.hypotheticalOrderedSetAggregates_windowEmulation();
+			queryEngine.getSqmFunctionRegistry().registerNamed(
+					"json_valid",
+					queryEngine.getTypeConfiguration()
+							.getBasicTypeRegistry()
+							.resolve( StandardBasicTypes.BOOLEAN )
+			);
+			if ( getVersion().isSameOrAfter( 10, 3, 3 ) ) {
+				commonFunctionFactory.inverseDistributionOrderedSetAggregates_windowEmulation();
+			}
+		}
+	}
+
+	@Override
+	public SqlAstTranslatorFactory getSqlAstTranslatorFactory() {
+		return new StandardSqlAstTranslatorFactory() {
+			@Override
+			protected <T extends JdbcOperation> SqlAstTranslator<T> buildTranslator(
+					SessionFactoryImplementor sessionFactory, Statement statement) {
+				return new MariaDBLegacySqlAstTranslator<>( sessionFactory, statement );
+			}
+		};
+	}
+
+	@Override
+	public boolean supportsWindowFunctions() {
+		return getVersion().isSameOrAfter( 10, 2 );
+	}
+
+	@Override
```

Comment:
Looks like something got lost after the rebase:

```suggestion
	@Override
	public boolean supportsLateral() {
		// See https://jira.mariadb.org/browse/MDEV-19078
		return false;
	}

	@Override
	public boolean supportsRecursiveCTE() {
		return getVersion().isSameOrAfter( 10, 2 );
	}

	@Override
```

Reply:
I'm confused by this one; where does this change come from, it doesn't appear to be in the main version of MariaDBDialect (unless I've gone selectively blind)

Reply:
You're right, it should have been this:

```suggestion
	@Override
	public boolean supportsLateral() {
		// See https://jira.mariadb.org/browse/MDEV-19078
		return false;
	}

	@Override
```